providers/subclassing_an_existing_provider.md

```diff
@@ -0,0 +1,188 @@
+
+### Creating the new plugin
+
+First step is to create a new empty plugin:
+```bash
+$ bundle exec rails generate manageiq:provider ManageIQ::Providers::AwesomePrivateCloud --no-scaffolding --vcr
+** ManageIQ master, codename: Oparin
+      create  
+         run  git init /home/grare/adam/src/manageiq/manageiq/plugins/manageiq-providers-awesome_private_cloud from "."
+
+Initialized empty Git repository in /home/grare/adam/src/manageiq/manageiq/plugins/manageiq-providers-awesome_private_cloud/.git/
+      create  manageiq-providers-awesome_private_cloud.gemspec
+      create  .codeclimate.yml
+      create  .gitignore
+      create  .rspec
+      create  .rspec_ci
+      create  .rubocop.yml
+      create  .rubocop_cc.yml
+      create  .rubocop_local.yml
+      create  .whitesource
+      create  .yamllint
+      create  Gemfile
+      create  LICENSE.txt
+      create  Rakefile
+      create  README.md
+      create  bin/ci/after_script
+      create  bin/rails
+      create  bin/setup
+      create  bin/update
+      create  bundler.d
+      create  bundler.d/.keep
+      create  config/secrets.defaults.yml
+      create  config/settings.yml
+      create  lib/manageiq-providers-awesome_private_cloud.rb
+      create  lib/manageiq/providers/awesome_private_cloud/engine.rb
+      create  lib/manageiq/providers/awesome_private_cloud/version.rb
+      create  lib/tasks/README.md
+      create  lib/tasks_private/spec.rake
+      create  locale
+      create  locale/.keep
+      create  spec/factories
+      create  spec/support
+      create  spec/spec_helper.rb
+      insert  /home/grare/adam/src/manageiq/manageiq/Gemfile
+      create  spec/models/manageiq/providers/awesome_private_cloud
+        gsub  lib/tasks_private/spec.rake
+      insert  .yamllint
+      append  spec/spec_helper.rb
+
+```
+
+Follow the steps from [Creating a new provider](writing_a_new_provider.md) for setting up your new plugin for local development by adding to your `bundler.d/override.rb` and symlinking your `spec/manageiq` directory.
+
+You'll notice however that we chose "--no-scaffolding" option, this is going to allow us to define the classes that we need as subclasses of the OpenStack provider.
+
+### Creating the subclasses
+
+Now that we have an empty plugin we have to start adding the subclasses that we need for the provider to work.
+
+First let's create the initial directories that will hold our subclasses:
+```bash
+$ mkdir -p app/models/manageiq/providers/awesome_private_cloud/{cloud_manager,network_manager,storage_manager} app/models/manageiq/providers/awesome_private_cloud/inventory/{collector,parser,persister}
+```
+
+```ruby
```

Comment:
I think there's a sentence missing before this; the previous comment talks about directories.

Reply:
https://github.com/ManageIQ/guides/pull/492/files#diff-f19823aaa72a5e378b421bb97147c5ac793580e22c74efa57d92f72db211d581R72
got ahead of myself talking about sti

Reply:
I think it got lost somehow, and I don't see it in the diff?

Reply:
Yeah switched machines and didn't pull latest from this branch, fixed